pandas/core/base.py

```diff
@@ -780,7 +780,7 @@ def array(self) -> ExtensionArray:
 
         return result
 
-    def to_numpy(self, dtype=None, copy=False):
+    def to_numpy(self, dtype=None, copy=False, **kwargs):
```

Comment:
this is changing the signature of all EA, but are you adding tests? (e.g. StringArray / IntArray), or as followup?

Reply:
> this is changing the signature of all EA, but are you adding tests? (e.g. StringArray / IntArray), or as followup?

Note this is not changing the signature of the EA method (this is the Series/Index method)

I think in general it might be useful to pass through kwargs, in that way ExtensionArray authors can have more specific control over the conversion to numpy (if we do this, we should add a test for it for one of the test EAs, like DecimalArray, with an additional keyword in the `to_numpy` method).

Specifically for `na_value`, we could maybe add it to the actual signature. Although it is right now not implemented for most of the dtypes (only for boolean EA dtype).

Reply:
right, actually meant *that*. yeah I would be explict here I think. the question is can you? e.g.  you want the default to be ``libmissing.NA``, but that is *not* the default for anything but BA now? I think it would be better to be explicit here.

Reply:
Yes, that's what makes it more complicated here. It's really a EA-specific keyword, and the general solution for that is passing through keywords (as done here), I think. 
Of course, for our own EAs, we can make exceptions and explicitly add them.

Now, since the other dtypes don't yet implement this, the default of `libmissing.NA` might not necessarily be a problem (this default is not valid for other dtypes, but it's not used for those dtypes anyway). Now, having this keyword with such default might also be confusing, since the NA won't be used for most keywords. 

If we add it here explicitly, I would probably use the `lib._no_default` trick

Reply:
> If we add it here explicitly, I would probably use the lib._no_default trick

I think this would be the better long term soln ,yes?